pkg/config/config_test.go

```diff
@@ -897,6 +897,74 @@ image_copy_tmp_dir="storage"`
 			gomega.Expect(content.String()).To(gomega.Equal(""))
 			logrus.SetOutput(os.Stderr)
 		})
+
+		It("--connection flag precedence is more than env variables", func() {
+			// test setup
+			config, err := New(nil)
+			gomega.Expect(err).To(gomega.BeNil())
+			gomega.Expect(config.Engine.ServiceDestinations).Should(gomega.BeEmpty())
+
+			// add connections
+			config.Engine.ActiveService = "QA"
+			config.Engine.ServiceDestinations = map[string]Destination{
+				"QB": {
+					URI:      "https://qb/run/podman/podman.sock",
+					Identity: "/.ssh/qb_id_rsa",
+				},
+				"QA": {
+					URI:      "https://qa/run/podman/podman.sock",
+					Identity: "/.ssh/id_rsa",
+				},
+			}
+
+			// write the modified config back to store
+			err = config.Write()
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+			// unset all necessary env variables for isolated testing
+			oldContainerHost, hostEnvSet := os.LookupEnv("CONTAINER_HOST")
+			os.Unsetenv("CONTAINER_HOST")
+			oldContainerSSH, sshEnvSet := os.LookupEnv("CONTAINER_SSHKEY")
+			os.Unsetenv("CONTAINER_SSHKEY")
+			oldContainerConn, conEnvSet := os.LookupEnv("CONTAINER_CONNECTION")
+			os.Unsetenv("CONTAINER_CONNECTION")
+
+			// call ActiveDestination without env variable set
+			u, i, m, err := config.ActiveDestination()
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			gomega.Expect(m).To(gomega.Equal(false))
+			gomega.Expect(u).To(gomega.Equal("https://qa/run/podman/podman.sock"))
+			gomega.Expect(i).To(gomega.Equal("/.ssh/id_rsa"))
+
+			// now add all necessary env variables
+			os.Setenv("CONTAINER_HOST", "foo.bar")
+			os.Setenv("CONTAINER_SSHKEY", "/.ssh/newid_rsa")
+			os.Setenv("CONTAINER_CONNECTION", "QB")
+
+			// call ActiveDestination with env variable set
+			u, i, m, err = config.ActiveDestination()
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			gomega.Expect(m).To(gomega.Equal(false))
+			gomega.Expect(u).To(gomega.Equal("https://qa/run/podman/podman.sock"))
+			gomega.Expect(i).To(gomega.Equal("/.ssh/id_rsa"))
+
+			// reset back to original state of env variables
+			if hostEnvSet {
+				os.Setenv("CONTAINER_HOST", oldContainerHost)
+			} else {
+				os.Unsetenv("CONTAINER_HOST")
+			}
+			if sshEnvSet {
+				os.Setenv("CONTAINER_SSHKEY", oldContainerSSH)
+			} else {
+				os.Unsetenv("CONTAINER_SSHKEY")
+			}
+			if conEnvSet {
+				os.Setenv("CONTAINER_CONNECTION", oldContainerConn)
+			} else {
+				os.Unsetenv("CONTAINER_CONNECTION")
+			}
```

Comment:
this needs to happen in a defer function directly after you read the envs. So basically juts out this block in a `defer func() {//code here}()` directly after your unsetenv calls on line 930.

If one of the Expect() calls above fails it will stop tight there this will never be executed.